docs/application/native/guides/compliations/watchface-complication-provider.md

```diff
@@ -0,0 +1,247 @@
+complication provider app notifies that the complication data is changed.
+
+```cpp
+void _watchface_complication_provider_update_requested_cb(const char *provider_id,
+		const char *req_appid, watchface_complication_type_e type, const bundle *context,
+		bundle *share_data, void *user_data)
+{
+}
+
+bool app_create(void *data)
+{
+	watchface_complication_provider_add_update_requested_cb("PROVIDER_ID",
+			_watchface_complication_provider_update_requested_cb, NULL);
+}
+```
+
+The complication provider app should be set the updated complication data to shared_data in the update requested callback.
+And it is possible to check the data is valid or not using watchface_complication_provider_data_is_valid() fundtion.
+If the data is invalid, watchface_complication_updated_cb of the complication will not be called.
+
+The complication data can be set using following APIs. These APIs can be used only in watchface_complication_provider_update_requested_cb().
+
+| API | Description |
+|--------|-----------------|
+| watchface_complication_provider_data_set_title() | Set the complication title |
+| watchface_complication_provider_data_set_short_text() | Set the short text type data |
+| watchface_complication_provider_data_set_long_text() | Set the long text type data |
+| watchface_complication_provider_data_set_image_path() | Set the image path type data |
+| watchface_complication_provider_data_set_ranged_value() | Set the ranged value type data |
+| watchface_complication_provider_data_set_icon_path() | Set the icon path type data |
+| watchface_complication_provider_data_set_timestamp() | Set the timestamp type data |
+| watchface_complication_provider_data_set_extra_data() | Set the extra data |
+
+
+```cpp
+void _watchface_complication_provider_update_requested_cb(const char *provider_id,
+		const char *req_appid, watchface_complication_type_e type, const bundle *context,
+		bundle *share_data, void *user_data)
+{
+	bool is_valid;
+
+	if (type == WATCHFACE_COMPLICATION_TYPE_SHORT_TEXT) {
+		watchface_complication_provider_data_set_short_text(shared_data, "updated short text");
+
+		watchface_complication_provider_data_is_valid(shared_data, &is_valid);
+	}
+}
+```
+
+If the callback is not nessary, it can be removed by using 
+watchface_complication_provider_remove_update_request_cb().
+
+```cpp
+{
+	watchface_complication_provider_remove_update_requested_cb("PROVIDER_ID",
+			_watchface_complication_provider_update_requested_cb);
+}
+```
+
+# Notify the Complication Data Update
+
+When the complication data is changed, the complication provider app can notify that the complication data is updated.
+If the notification is successfully transfer to the complication, the complication data update request will be occured.
+
+
+```cpp
+{
+	watchface_complication_provider_notify_update("PROVIDER_ID");
+}
+```
+
+# Transfered Event Action
+
+Complication can provide additional action for touching when the complication is touched.

```

Comment:
Anything that complication provider app wants. It's defined by the complication provider application. If the complication provider app has a setting app for complication and wants to launch when touched, it's possible. If complication app has no action for touching, it can be ignored.

Reply:
Okay, thanks for explanation
Please can you rephrased as:
The complication can provide additional action that is touching.